p2p/src/main/kotlin/maru/p2p/MaruPeerManager.kt

```diff
@@ -44,14 +52,18 @@ class MaruPeerManager(
     }, 20000, 20000, TimeUnit.MILLISECONDS)
   }
 
+  val blocksBehindThreshold: ULong = 32uL
+
   private val log: Logger = LogManager.getLogger(this.javaClass)
   private val maxPeers = p2pConfig.maxPeers
   private val currentlySearching = AtomicBoolean(false)
-
   private val connectionInProgress = mutableListOf<Bytes>()
   private var discoveryService: MaruDiscoveryService? = null
+  private val statusExchangingMaruPeers = mutableMapOf<NodeId, MaruPeer>()
```

Comment:
What is the difference between `statusExchangingMaruPeers` and `connectedPeers`?

Reply:
peers are added to the statusExchanginMaruPeers when they have just connected and about to exchange the status message. To allow us to send and receive the status message we have to be able to find these by peer id. Once the statusExchangingPeers have been checked in onConnect() they are added to connectedPeers. connectedPeers are the peers that can be used to send requests.